src/coreclr/jit/llvm.cpp

```diff
@@ -2713,6 +2754,34 @@ void Llvm::lowerToShadowStack()
     }
 }
 
+void Llvm::LowerPromotedFields()
```

Comment:
I don't think this needs to be a separate phase, it can be part of the general lowering (what is now called `lowerToShadowStack`, but should be renamed to just "Lower", I suppose).